test/flows/find_coronavirus_support_flow_test.rb

```diff
@@ -0,0 +1,597 @@
+        assert_next_node :self_employed, for_response: %w[being_unemployed mental_health]
+      end
+    end
+  end
+
+  context "question: feel_unsafe" do
+    setup do
+      testing_node :feel_unsafe
+      add_responses need_help_with: %w[feeling_unsafe]
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    %w[yes concerned_about_others no not_sure].each do |response|
+      should "have a next node of nation if response is #{response} and there aren't any other needs_help_with questions" do
+        assert_next_node :nation, for_response: response
+      end
+
+      should "have next node of next question group for #{response} if there are other needs_help_with questions" do
+        add_responses need_help_with: %w[feeling_unsafe mental_health]
+
+        assert_next_node :mental_health_worries, for_response: response
+      end
+    end
+  end
+
+  context "question: afford_rent_mortgage_bills" do
+    setup do
+      testing_node :afford_rent_mortgage_bills
+      add_responses need_help_with: %w[paying_bills]
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    %w[yes no not_sure].each do |response|
+      should "have a next node of nation if response is #{response} and there aren't any other needs_help_with questions" do
+        assert_next_node :nation, for_response: response
+      end
+
+      should "have next node of next question group for #{response} if there are other needs_help_with questions" do
+        add_responses need_help_with: %w[paying_bills mental_health]
+
+        assert_next_node :mental_health_worries, for_response: response
+      end
+    end
+  end
+
+  context "question: afford_food" do
+    setup do
+      testing_node :afford_food
+      add_responses need_help_with: %w[getting_food]
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    %w[yes no not_sure].each do |response|
+      should "have a next node of get_food if response is #{response}" do
+        assert_next_node :get_food, for_response: response
+      end
+    end
+  end
+
+  context "question: get_food" do
+    setup do
+      testing_node :get_food
+      add_responses need_help_with: %w[getting_food],
+                    afford_food: "yes"
+    end
+
+    should "render the question" do
+      assert_rendered_question
+    end
+
+    %w[yes no not_sure].each do |response|
+      should "have a next node of nation if response is #{response} and there aren't any other needs_help_with questions" do
+        assert_next_node :nation, for_response: response
+      end
+
+      should "have next node of next question group for #{response} if there are other needs_help_with questions" do
+        add_responses need_help_with: %w[getting_food mental_health]
+
+        assert_next_node :mental_health_worries, for_response: response
+      end
+    end
+  end
+
+  context "question: self_employed" do
```

Comment:
Do you think it'd be worthwhile to add an extra context block for `being unemployed` that wraps around this? Figure it would give a bit more context around how the code is structured which might be useful if any new folk come into this codebase. Also applies to the other groups

Reply:
Sounds good to me. @kevindew How does this suggestion fit with the convention of the new test structure?

Reply:
It's unfortunately counter to the approach of structuring these flow tests and I don't think you should do it.

I can see the logic of the suggestion, these questions are both related to a journey through this of unemployment, however I think that is just the tip of the iceberg for Smart Answers as many nodes are only related to journeys of other ones. I think the only consistent logical approach is that given nodes are a top level concept then they should only be tested as top level concepts.

I don't see the benefit of that myself. Both self_employed and have_you_been_made_unemployed are top level nodes so I don't see why they should be nested. 

That seems like a slippery slope towards context based on user journeys that can easily multiply into the thousands.